File: 이것이 취업을 위한 코딩 테스트다/8-2.py
# 피보나치 수열 (재귀적), p.212
# 한 번 계산된 결과를 메모이제이션하기 위한 리스트 초기화
d = [0] * 100

# 피보나치 함수를 재귀함수로 구현 (탑다운)
def fibo(x):
    if x==1 or x==2 :
        return 1
    if d[x] != 0 :
        return d[x]
    d[x] = fibo(x-1) + fibo(x-2)
    return d[x]


# 피보나치 수열 (반복적)
d = [0] * 100

d[1] = 1
d[2] = 1
n = 99

for i in range(3, n+1):
    d[i] = d[i-1] + d[i-2]

File: 이것이 취업을 위한 코딩 테스트다 (JAVA)/8-2.java
import java.util.*;

public class Main {
    public static long[] arr = new long[100];

    public static long fibo(int x) {
        if (x == 1 || x == 2) {
            return 1;
        }

        if (arr[x] != 0) {
            return arr[x];
        }
        
        arr[x] = fibo(x-1) + fibo(x-2);
        return arr[x];
    }

    public static void main(String[] args) {
        System.out.println(fibo(50));
    }
}